app/controllers/cloud_subnet_controller.rb

```diff
@@ -52,17 +52,38 @@ def cloud_subnet_form_fields
     }
   end
 
+  def cloud_subnet_networks_by_ems
+    assert_privileges("cloud_subnet_new")
+    networks = []
+    available_networks = CloudNetwork.where(:ems_id => params[:id]).find_each
+    available_networks.each do |network|
+      networks << { 'name' => network.name, 'id' => network.id }
+    end
+    render :json => {
+      :available_networks => networks
+    }
+  end
+
+  def cloud_tenants_by_ems
+    assert_privileges("cloud_subnet_new")
+    tenants = []
+    available_tenants = CloudTenant.where(:ems_id => params[:id]).find_each
+    available_tenants.each do |tenant|
+      tenants << { 'name' => tenant.name, 'id' => tenant.id }
+    end
+    render :json => {
+      :available_tenants => tenants
+    }
+  end
+
   def new
     assert_privileges("cloud_subnet_new")
     @subnet = CloudSubnet.new
     @in_a_form = true
     @network_provider_choices = {}
-    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each { |ems| @network_provider_choices[ems.name] = ems.id }
-    # TODO: (gildub) Replace with angular lookup to narrow choice dynamically
-    @network_choices = {}
-    CloudNetwork.all.each { |network| @network_choices[network.name] = network.ems_ref }
-    @cloud_tenant_choices = {}
-    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
```

Comment:
This is a providers plug-ability concern, in general we should not have ManageIQ::Providers::Openstack anywhere in the classic-ui. The filtering should be done using supports? on the classes.

But since this was already here, I am fine with a separate PR fix, but it can be fixed here, since it relates to the PR, so it's your call :-)